app/services/settings/upsert.rb

```diff
@@ -30,7 +30,7 @@ def upsert_settings
           settings_class.public_send("#{key}=", value.reject(&:blank?))
         elsif value.respond_to?(:to_h) && value.present?
           settings_class.public_send("#{key}=", value.to_h)
-        elsif value.present?
+        else
```

Comment:
This was here to avoid setting the values to empty strings, which prevents them from returning their default values:

```ruby
Settings::General.app_domain
#=> "localhost:3000" (default value)
Settings::General.app_domain = ""
Settings::General.app_domain
#=> ""
```
This can also be seen in the spec you wrote and while it may not be a problem for `Settings::SMTP.address` it might be for settings where we expect a fallback to the default value.

Reply:
We probably never want to add an empty string here, so maybe this is a sensible workaround (we already know we are not dealing with array or hash values here):

```ruby
elsif value.blank?
  settings_class.public_send("#{key}=", nil)
```

Reply:
Oof, I overlooked that. Thank you for the suggestion!